xbmc/cores/VideoPlayer/VideoPlayerRadioID3.h

```diff
@@ -0,0 +1,56 @@
+/*
+ *  Copyright (C) 2005-2022 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include "DVDMessageQueue.h"
+#include "FileItem.h"
+#include "IVideoPlayer.h"
+#include "threads/CriticalSection.h"
+#include "threads/Thread.h"
+
+#include <memory>
+
+#include <taglib/id3v1tag.h>
+#include <taglib/id3v2tag.h>
+#include <taglib/tstringlist.h>
```

Comment:
A far as I can see you only use pointers and refs to the taglib data types, thus all of them can be forward-decalred here.

Reply:
This got a bit awkward looking to me and required the header to have what seems like a bit too much "intimate knowledge" of TagLib. I've made the change and it builds fine, but want to check that you are good with the end result (below, and already clang-formatted) as opposed to including the headers.

My only (minor) concern, is that doing it this way might infer a bit too much "intimate" knowledge of TagLib's implementation details and leave us open to breaking changes from something TagLib considered to be opaque. I doubt that would ever happen, and of course updating a third-party dependency like TagLib implies that a certain amount of due diligence would be executed to fix anything that went wrong.

Below replacement code is in upcoming force-push, if it's no good LMK and I can change it again no problem:

```
namespace TagLib
{
template<class T>
class List;
class StringList;
namespace ID3v1
{
class Tag;
}
namespace ID3v2
{
class Frame;
typedef TagLib::List<Frame*> FrameList;
class Tag;
} // namespace ID3v2
} // namespace TagLib
```

Reply:
Prefer the C++ `using` instead of the C `typedef`. But my thoughts are that maybe the headers are preferred. @ksooo ?

Reply:
switching over to using vs. typedef built fine for me on Windows (VS 2022):

```
namespace TagLib
{
template<class T>
class List;
class StringList;
namespace ID3v1
{
class Tag;
}
namespace ID3v2
{
class Frame;
using FrameList = TagLib::List<Frame*>;
class Tag;
} // namespace ID3v2
} // namespace TagLib
```

I am good with whatever final decision is made (forward declaration vs. headers and/or using vs. typedef in the former case). Whatever builds the cleanest on all platforms!

Reply:
> But my thoughts are that maybe the headers are preferred. @ksooo ?

 Yes, let's keep the includes in this case.